src/web100srv.c

```diff
@@ -1527,647 +1326,364 @@ void child_process(int parent_pipe, SSL_CTX *ssl_context, int ctlsockfd) {
-    // tells the client that the applet should drop out of the wait
-    // loop and get ready to begin testing.  The write() on the pipe
-    // tells the child process it's at the head of the queue, or it's
-    // in multi-client mode, and its OK for this client process to
-    // wake up and begin testing.
-    //
-    //  This is the point in time where we can/should contact any meta
-    //  scheduler to verify that it is OK to really start a test.  If
-    //  we are sharing the server host with other applications, and we
-    //  want unfettered access to the link, now is the time to make
-    //  this request.
-
-    dispatch_client:
-      memset(tmpstr, 0, sizeof(tmpstr));
-      if (multiple == 1) {
-        if (mchild == NULL) mchild = head_ptr;
-        if (mchild->running == 1) continue;
-        log_println(3, "New mclient '%d'(%d) asking for service", mclients,
-                    mchild->pid);
-        mchild->stime = time(0);
-        mchild->running = 1;
-        mclients++;
-        log_println(5, "sending 'GO' signal to client");
-        for (i = 0; i < 5; i++) {
-          child_signal = SRV_QUEUE_TEST_STARTS_NOW;
-          retcode = write(mchild->pipe, &child_signal, sizeof(int));
-          log_println(6, "write(%d) returned %d, errno=%d", mchild->pid,
-                      retcode, errno);
-          if (retcode == sizeof(int)) break;
-          if ((retcode == -1) && (errno == EINTR)) continue;
-          log_println(5,
-                      "Failed to write 'GO' message to client %d, "
-                      "reason=%d, errno=%d",
-                      mchild->pid, retcode, errno);
-          // TODO: handle other error conditions
-          if (retcode == -1) {
-            log_println(1, "Dispatch multi-client failed because '%s'",
-                        strerror(errno));
-            close(mchild->pipe);
-            kill(chld_pid, SIGTERM);
-            goto mainloop;
-          }
-        }
-        close(mchild->pipe);
-      } else {  // multiple client loop complete
-        head_ptr->stime = time(0);
-        head_ptr->running = 1;
-        log_println(5, "sending 'GO' signal to client");
-        for (i = 0; i < 5; i++) {
-          child_signal = SRV_QUEUE_TEST_STARTS_NOW;
-          retcode = write(mchild->pipe, &child_signal, sizeof(int));
-          if (retcode == sizeof(int)) break;
-          if ((retcode == -1) && (errno == EINTR)) continue;
-          // TODO: handle other error conditions
-          if (retcode == -1) {
-            log_println(1, "Dispatch multi-client failed because '%s'",
-                        strerror(errno));
-            close(head_ptr->pipe);
-            kill(chld_pid, SIGTERM);
-            goto mainloop;
-          }
-        }
-        close(head_ptr->pipe);
-      }
-    }
+  SSL_library_init();
+  OpenSSL_add_all_algorithms();
+  SSL_load_error_strings();
+  ssl_context = SSL_CTX_new(SSLv23_server_method());
```

Comment:
`SSLv23_server_method` is marked as deprecated: https://www.openssl.org/docs/ssl/SSL_CTX_new.html